WordPress/Classes/Services/PostTagService.m

```diff
@@ -118,16 +120,38 @@ - (void)searchTagsWithName:(NSString *)nameQuery
 
 - (void)deleteTag:(PostTag*)tag
           forBlog:(Blog *)blog
+          success:(nullable void (^)(void))success
+          failure:(nullable void (^)(NSError *error))failure
 {
-    //Not implemented yet
-    NSLog(@"Delete tag %@ for blog %@", tag, blog);
+    NSObject<TaxonomyServiceRemote> *remote = [self remoteForBlog:blog];
+
+    RemotePostTag *remoteTag = [self remoteTagWith:tag];
+
+    [remote deleteTag:remoteTag success:^(RemotePostTag * _Nonnull remoteTag) {
+        [self.managedObjectContext deleteObject:tag];
+        [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+        if (success) {
+            success();
+        }
+    } failure:failure];
```

Comment:
It would be nice if we could call the `handleError:forBlog:withFailure` to log the error description.